ocp_utilities/infra.py

```diff
@@ -41,3 +46,117 @@ def assert_nodes_schedulable(nodes):
         raise NodeUnschedulableError(
             f"Following nodes are in unscheduled state: {unschedulable_nodes}"
         )
+
+
+class DynamicClassCreator:
```

Comment:
This is infra component, but for ease of use, we might want to consider having a separate file for this one. It would reduce the possibility of circular import as this repo grows.

Reply:
`data_collector` no longer imports this module